pkg/admission/prep_downscale.go

```diff
@@ -35,10 +37,15 @@ const (
 	maxPrepareGoroutines        = 32
 )
 
-func PrepareDownscale(ctx context.Context, logger log.Logger, ar admissionv1.AdmissionReview, api *kubernetes.Clientset, useZoneTracker bool, zoneTrackerConfigMapName string) *admissionv1.AdmissionResponse {
+func PrepareDownscale(ctx context.Context, namespace string, logger log.Logger, ar admissionv1.AdmissionReview, api *kubernetes.Clientset, useZoneTracker bool, zoneTrackerConfigMapName string, metrics *metrics.Metrics) *admissionv1.AdmissionResponse {
+	rt := http.DefaultTransport
+	if namespace != "" {
+		// HTTP client side cluster validation.
+		rt = middleware.ClusterValidationRoundTripper(rt, namespace, logger, metrics)
```

Comment:
cluster protection seems a bit out of scope for the admission webhooks.

Can we instead inject a roundtripper? That way we can keep the cluster_verification roundtripper only in main.go and not have to create a special package for it either.

Reply:
i think metrics wouldn't need to be exported either if we don't need to instantiate the middleware outside of cmd/rollout-operator

Reply:
PTAL at my latest revision.